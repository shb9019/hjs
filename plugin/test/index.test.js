import path from 'path';
import pluginTester from 'babel-plugin-tester';
import hjsPlugin from '../src';

/**
 * 1. [X] Simple unchanged
 * 2. [X] 2 Parameters basic function
 * 3. [X] 3 Parameters basic function
 * 4. [X] 3 Parameters function expression
 * 5. [X] 3 Parameters anonymous function expresion
 * 6. [X] 3 Parameters arrow functions
 * 7. [X] Default parameters
 * 8. [X] IIFE
 * 9. Variadic functions and rest parameters
 * 10. Async functions
 * 11. Changing scope
 * 12. Optional parameters
 * 13. Decorators
 * 14. Ensure scope is respected if it is passed through call, bind and apply.
 * 15. Generator functions
 * 16. Figure out directives
 * 17. Arrow functions within in-built functions
 * 18. Dynamically generated functions
 */

pluginTester({
    plugin: hjsPlugin,
    filename: __filename,
    fixtures: path.join(__dirname, '__fixtures__'),
    tests: [
        {
            fixture: '__fixtures__/default-param-test.js',
            snapshot: true,
        },
        {
            fixture: '__fixtures__/partial-default-param-test.js',
            snapshot: true,
        },
        {
            fixture: '__fixtures__/iife-test.js',
            snapshot: true,
        },
        {
            fixture: '__fixtures__/rest-param-test.js',
            snapshot: true,
        },
    ]
});