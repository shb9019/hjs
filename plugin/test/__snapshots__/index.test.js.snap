// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

function withDefaults(a, b, c = 1, d = 2, e = 3) {
  return [a, b, c, d, e];
}

      ↓ ↓ ↓ ↓ ↓ ↓

function withDefaults(a, b, c = 1, d = 2, e = 3) {
  let __$curriedResponse = function (a) {
    return function (b, c = 1, d = 2, e = 3) {
      let __$curriedResponse = function (b) {
        return function (c = 1, d = 2, e = 3) {
          let __$curriedResponse = function (c = 1) {
            return function (d = 2, e = 3) {
              let __$curriedResponse = function (d = 2) {
                return function (e = 3) {
                  return [a, b, c, d, e];
                };
              };

              for (const arg of arguments) {
                __$curriedResponse = __$curriedResponse(arg);
              }

              if (arguments.length === 0)
                __$curriedResponse = __$curriedResponse(undefined);
              return __$curriedResponse;
            };
          };

          for (const arg of arguments) {
            __$curriedResponse = __$curriedResponse(arg);
          }

          if (arguments.length === 0)
            __$curriedResponse = __$curriedResponse(undefined);
          return __$curriedResponse;
        };
      };

      for (const arg of arguments) {
        __$curriedResponse = __$curriedResponse(arg);
      }

      if (arguments.length === 0)
        __$curriedResponse = __$curriedResponse(undefined);
      return __$curriedResponse;
    };
  };

  for (const arg of arguments) {
    __$curriedResponse = __$curriedResponse(arg);
  }

  if (arguments.length === 0)
    __$curriedResponse = __$curriedResponse(undefined);
  return __$curriedResponse;
}


`;

exports[`unknown plugin 2. unknown plugin: 2. unknown plugin 1`] = `

function withDefaults(a, b, c = 1, d = 2, e = 3, f, g) {
  return [a, b, c, d, e, f, g];
}

      ↓ ↓ ↓ ↓ ↓ ↓

function withDefaults(a, b, c = 1, d = 2, e = 3, f, g) {
  let __$curriedResponse = function (a) {
    return function (b, c = 1, d = 2, e = 3, f, g) {
      let __$curriedResponse = function (b) {
        return function (c = 1, d = 2, e = 3, f, g) {
          let __$curriedResponse = function (c = 1) {
            return function (d = 2, e = 3, f, g) {
              let __$curriedResponse = function (d = 2) {
                return function (e = 3, f, g) {
                  let __$curriedResponse = function (e = 3) {
                    return function (f, g) {
                      let __$curriedResponse = function (f) {
                        return function (g) {
                          return [a, b, c, d, e, f, g];
                        };
                      };

                      for (const arg of arguments) {
                        __$curriedResponse = __$curriedResponse(arg);
                      }

                      if (arguments.length === 0)
                        __$curriedResponse = __$curriedResponse(undefined);
                      return __$curriedResponse;
                    };
                  };

                  for (const arg of arguments) {
                    __$curriedResponse = __$curriedResponse(arg);
                  }

                  if (arguments.length === 0)
                    __$curriedResponse = __$curriedResponse(undefined);
                  return __$curriedResponse;
                };
              };

              for (const arg of arguments) {
                __$curriedResponse = __$curriedResponse(arg);
              }

              if (arguments.length === 0)
                __$curriedResponse = __$curriedResponse(undefined);
              return __$curriedResponse;
            };
          };

          for (const arg of arguments) {
            __$curriedResponse = __$curriedResponse(arg);
          }

          if (arguments.length === 0)
            __$curriedResponse = __$curriedResponse(undefined);
          return __$curriedResponse;
        };
      };

      for (const arg of arguments) {
        __$curriedResponse = __$curriedResponse(arg);
      }

      if (arguments.length === 0)
        __$curriedResponse = __$curriedResponse(undefined);
      return __$curriedResponse;
    };
  };

  for (const arg of arguments) {
    __$curriedResponse = __$curriedResponse(arg);
  }

  if (arguments.length === 0)
    __$curriedResponse = __$curriedResponse(undefined);
  return __$curriedResponse;
}


`;

exports[`unknown plugin 3. unknown plugin: 3. unknown plugin 1`] = `

const result = (function (a, b, c) {
  return a + b + c;
})(1, 2, 3);

      ↓ ↓ ↓ ↓ ↓ ↓

const result = (function (a, b, c) {
  let __$curriedResponse = function (a) {
    return function (b, c) {
      let __$curriedResponse = function (b) {
        return function (c) {
          return a + b + c;
        };
      };

      for (const arg of arguments) {
        __$curriedResponse = __$curriedResponse(arg);
      }

      if (arguments.length === 0)
        __$curriedResponse = __$curriedResponse(undefined);
      return __$curriedResponse;
    };
  };

  for (const arg of arguments) {
    __$curriedResponse = __$curriedResponse(arg);
  }

  if (arguments.length === 0)
    __$curriedResponse = __$curriedResponse(undefined);
  return __$curriedResponse;
})(1, 2, 3);


`;

exports[`unknown plugin 4. unknown plugin: 4. unknown plugin 1`] = `

const multiplyAll = (a, b, ...rest) => {
  let product = a * b;
  for (const item of rest) {
    product = product * item;
  }
  return product;
};

      ↓ ↓ ↓ ↓ ↓ ↓

const multiplyAll = (a, b, ...rest) => {
  let __$curriedResponse = function (a) {
    return function (b, ...rest) {
      let __$curriedResponse = function (b) {
        return function (...rest) {
          let product = a * b;

          for (const item of rest) {
            product = product * item;
          }

          return product;
        };
      };

      for (const arg of arguments) {
        __$curriedResponse = __$curriedResponse(arg);
      }

      if (arguments.length === 0)
        __$curriedResponse = __$curriedResponse(undefined);
      return __$curriedResponse;
    };
  };

  const args = [];
  if (a !== undefined) args.push(a);
  if (b !== undefined) args.push(b);
  if (undefined !== undefined) args.push(undefined);

  for (const arg of args) {
    __$curriedResponse = __$curriedResponse(arg);
  }

  if (args.length === 0) __$curriedResponse = __$curriedResponse(undefined);
  return __$curriedResponse;
};


`;
